repose-aggregator/components/services/open-tracing-service/src/test/scala/org/openrepose/core/services/opentracing/OpenTracingServiceImplTest.scala

```diff
@@ -0,0 +1,313 @@
+    tracer = mock[DelegatingTracer]
+
+    openTracingService = new OpenTracingServiceImpl(configurationService, tracer)
+  }
+
+  describe("init") {
+    it("should subscribe an opentracing configuration listener") {
+      openTracingService.init()
+
+      verify(configurationService).subscribeTo(
+        isEq("open-tracing.cfg.xml"),
+        any[URL](),
+        isA(classOf[UpdateListener[OpenTracingConfig]]),
+        isA(classOf[Class[OpenTracingConfig]]))
+    }
+  }
+
+  describe("destroy") {
+    it("should unsubscribe an opentracing configuration listener") {
+      openTracingService.destroy()
+
+      verify(configurationService).unsubscribeFrom(
+        isEq("open-tracing.cfg.xml"),
+        isA(classOf[UpdateListener[OpenTracingConfig]]))
+    }
+  }
+
+  describe("itInitialized") {
+    it("should return false if the service has not been configured") {
+      openTracingService.isInitialized shouldBe false
+    }
+
+    it("should return false if the service has been configured") {
+      openTracingService.configurationUpdated(
+        minimalOpenTracingConfig()
+      )
+
+      openTracingService.isInitialized shouldBe true
+    }
+  }
+
+  describe("getJaegerSamplerConfiguration") {
+    it("should return constant sampler set to on") {
+      val config = new JaegerTracerConfig()
+        .withJaegerSampling(new JaegerSamplingConstant()
+          .withToggle(Toggle.ON))
+
+      val samplerConfig = openTracingService.getJaegerSamplerConfiguration(config)
+
+      samplerConfig.getType shouldBe ConstSampler.TYPE
+      samplerConfig.getParam shouldEqual 1
+    }
+
+    it("should return constant sampler set to off") {
+      val config = new JaegerTracerConfig()
+        .withJaegerSampling(new JaegerSamplingConstant()
+          .withToggle(Toggle.OFF))
+
+      val samplerConfig = openTracingService.getJaegerSamplerConfiguration(config)
+
+      samplerConfig.getType shouldBe ConstSampler.TYPE
+      samplerConfig.getParam shouldEqual 0
+    }
+
+    it("should return rate limiting sampler set to value") {
+      val max = 2.0
+      val config = new JaegerTracerConfig()
+        .withJaegerSampling(new JaegerSamplingRateLimiting()
+          .withMaxTracesPerSecond(max))
+
+      val samplerConfig = openTracingService.getJaegerSamplerConfiguration(config)
+
+      samplerConfig.getType shouldBe RateLimitingSampler.TYPE
+      samplerConfig.getParam shouldEqual max
+    }
+
+    it("should return probabilistic sampler set to value") {
+      val probability = .2
+      val config = new JaegerTracerConfig()
+        .withJaegerSampling(new JaegerSamplingProbabilistic()
+          .withProbability(probability))
+
+      val samplerConfig = openTracingService.getJaegerSamplerConfiguration(config)
+
+      samplerConfig.getType shouldBe ProbabilisticSampler.TYPE
+      samplerConfig.getParam shouldEqual probability
+    }
+  }
+
+  describe("getJaegerSenderConfiguration") {
```

Comment:
Why are we testing the XJB Bindings/XJC generated code here?

Reply:
Wouldn't they make more sense in the Schema test?

Reply:
This is actually testing ever-so-slightly more than just the JAXB generated objects. It's testing the translation from our configuration object (which is a JAXB object) to a Jaeger sampler configuration. These tests are just to verify that configuration is being applied in the right places.